src/main/java/com/sthanik/expensetrack/dao/impl/CategoryRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.sthanik.expensetrack.dao.impl;
+
+import com.sthanik.expensetrack.dao.CategoryRepository;
+import com.sthanik.expensetrack.dao.DBConnection;
+import com.sthanik.expensetrack.model.CategoryDetails;
+import com.sthanik.expensetrack.model.UserDetails;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@Repository
+public class CategoryRepositoryImpl implements CategoryRepository {
+    @Override
+    public CategoryDetails getCategory(int categoryId) {
+        String getUser = "SELECT category_id,category_name,details \n" +
+                "FROM category\n" +
+                "WHERE category_id = ?\n" +
+                ";";
+        Connection connection = DBConnection.getDBConnection();
```

Comment:
what if getDBConnection throws an exception or fails to connect with DB and returns null?